addons/udes_suggest_location/tests/test_stock_move_line.py

```diff
@@ -105,48 +100,44 @@ def test_suggest_locations_errors_with_no_location_policy(self):
-        """Get the suggest locations via self for by product policy, looping through all
-        drop location constraints for completeness
-        """
-        #  Get locations to drop off apples
-        for drop_constraint in ("dont_scan", "scan", "suggest", "enforce"):
-            self.picking_type_pick.u_drop_location_constraint = drop_constraint
-            locs = self.apple_mls.suggest_locations()
-            self.assertEqual(locs, self.pick_apple_locs)
-        # Add u_drop_location_constraint with empty
-        for drop_constraint in ("suggest_with_empty", "enforce_with_empty"):
-            self.picking_type_pick.u_drop_location_constraint = drop_constraint
-            locs = self.apple_mls.suggest_locations()
-            self.assertEqual(locs, self.pick_all_locs)
-
-    def test_suggest_locations_via_picking_and_values_by_product(self):
-        """Get the suggest locations by picking and values, not self
-        This is set up for 'by_product' policy and loop through all u_drop_location_constraint
-        for completeness
-        """
-        #  Get locations to drop off apples
-        values = {"product_id": self.apple.id, "picking_id": self.picking_pick.id}
-        for drop_constraint in ("dont_scan", "scan", "suggest", "enforce"):
-            self.picking_type_pick.u_drop_location_constraint = drop_constraint
-            locs = self.MoveLine.suggest_locations(picking=self.picking_pick, values=values)
-            self.assertEqual(locs, self.pick_apple_locs)
-        # Add u_drop_location_constraint with empty
-        for drop_constraint in ("suggest_with_empty", "enforce_with_empty"):
-            self.picking_type_pick.u_drop_location_constraint = drop_constraint
-            locs = self.MoveLine.suggest_locations(picking=self.picking_pick, values=values)
-            self.assertEqual(locs, self.pick_all_locs)
-
     def test_suggest_locations_limit_results(self):
         """Get the suggest locations via self and limit the results"""
         # Set u_drop_location_constraint
         self.picking_type_pick.u_drop_location_constraint = "suggest_with_empty"
-        self.assertEqual(self.apple_mls.suggest_locations(), self.pick_all_locs)
+        self.assertEqual(self.apple_mls.suggest_locations(), self.pick_empty_locs)
         # Limit the results
         locs = self.apple_mls.suggest_locations(limit=1)
         self.assertEqual(len(locs), 1)
         self.assertIn(locs, self.pick_all_locs)
 
+    def test_create_mls_when_no_policy_set(self):
+        """Check that a pick can be created without a suggested locations policy"""
+        # Create picking
+        pick = self.create_picking(
+            self.picking_type_internal,
+            products_info=[{"product": self.apple, "qty": 5}],
+            confirm=True,
+            assign=True,
+        )
+        # Check mls has correct product
+        self.assertEqual(pick.move_line_ids.product_id, self.apple)
+
+    @unittest.skip("Fails for by_empty_location")
```

Comment:
Do we have stock in `self.test_goodsout_location_02`? At the moment I don't think we have set the test up to produce an invalid destination location when setting the policy to `by_empty_location`, as `self.test_goodsout_location_02` is actually empty and therefore a suggestible location

Reply:
There's no stock there, but I'm surprised that we can suggest a location that is not a child of `picking_type.default_location_dest_id`.

Reply:
yeah, I am not sure if there will be situation when we don't want that condition to be true. Maybe an issue does need to be created for it? It should be a simple change I think, just adding another check in `validate_location_dest()`

Reply:
When creating the picking we can set any destination location, `picking_type.default_location_dest_id` is only in case `location_dest_id` is not set. Therefore, `test.test_goodsout_location_02` is (in general) a valid destination location for any new picking regarding its picking type.

Reply:
Fixed by passing a destination location that already contains stock.